js/checkbox-wikirule.js

```diff
@@ -79,50 +113,80 @@ exports.parse = function() {
                 this.parser.source.substring(startPos + 4, this.parser.pos),
                 {parseAsInline: true});
 
-        // Put the listitem body in a span for easy reference
-        var itembody = {
-            type: "element",
-            tag: "span",
-            attributes: {
-                class: {type: "string", value: "checklistitem-body"}
-            },
-            children: parseResults.tree
-        };
-
+        // Use the listitem body as a label for the checkbox to get better accessibility
         var checkbox = {
             type: "element",
             tag: "input",
             attributes: {
                 type: {type: "string", value: "checkbox"},
-                pos: {type: "string", value: match.index}
+                pos: {type: "string", value: match.index},
+                id: {type: "string", value: match.index}
             }
         };
         if (match[1] === "x" || match[1] === "X") {
             checkbox.attributes.checked = {type: "boolean", value: true};
-            if (this.shouldStrikeChecked()) {
-                itembody.attributes.class.value += ", checkedchecklistitem-body";
-            }
         }
 
-        var removeicon = {
+        var itembody = {
+            type: "element",
+            tag: "label",
+            attributes: {
+                class: {type: "string", value: "checklistitem-body"},
+                for: {type: "string", value: match.index}
+            },
+            children: parseResults.tree
+        };
+
+        // Make a button to delete the item
+        // TODO: use svg bin icon, pattern as follows?
```

Comment:
Can you please pull this TODO out of the code and open a new issue with this content?